tauri/src/live_networks_listener/mod.rs

```diff
@@ -0,0 +1,67 @@
+mod error;
+
+use crate::db::DB;
+use crate::types::AlchemyResponse;
+use crate::{
+    types::GlobalState,
+    wallets::{WalletControl, Wallets},
+};
+pub use error::{Error, Result};
+use serde_json::json;
+use url::Url;
+
+#[derive(Debug)]
+pub struct LiveNetworksListener {
+    chain_id: u32,
+    db: DB,
+    alchemy_url: Url,
+}
+
+impl LiveNetworksListener {
+    pub fn new(chain_id: u32, db: DB, alchemy_url: Url) -> Self {
+        Self {
+            chain_id,
+            db,
+            alchemy_url,
+        }
+    }
+
+    pub fn run(&mut self) -> Result<()> {
+        let alchemy_url = self.alchemy_url.clone();
+        let db = self.db.clone();
+        let chain_id = self.chain_id;
+        tokio::spawn(async move { process(alchemy_url, db, chain_id).await });
+        Ok(())
+    }
+
+    pub fn stop(&mut self) {}
+}
+
+async fn process(ws_url: Url, db: DB, chain_id: u32) -> Result<()> {
+    let address = Wallets::read()
+        .await
+        .get_current_wallet()
+        .get_current_address()
+        .await;
+    let address = json!(address); // didnt find a better way to convert it to &str
```

Comment:
currently there's no way, but there should. easiest way is to just change the type:
```
pub struct ChecksummedAddress(Address);
```

into

```
pub struct ChecksummedAddress(pub Address);
```


then you can do `address.0`

---

but I think a better approach would be to have the listener receive the address when it starts (and we just restart to a new listener if the current account changes). that's a deeper change though